dwave/cloud/solver.py

```diff
@@ -1043,19 +1063,62 @@ def _sample(self, type_, linear, quadratic, offset, params,
 
         return computation
 
+    # kept for internal backwards compatibility and in case it's being
+    # used externally anywhere.
     def _format_params(self, type_, params):
         """Reformat some of the parameters for sapi."""
-        if 'initial_state' in params:
-            # NB: at this moment the error raised when initial_state does not match lin/quad (in
-            # active qubits) is not very informative, but there is also no clean way to check here
-            # that they match because lin can be either a list or a dict. In the future it would be
-            # good to check.
-            initial_state = params['initial_state']
-            if isinstance(initial_state, Mapping):
+        self.reformat_parameters(type_, params, self.properties, inplace=True)
+
+    @staticmethod
+    def reformat_parameters(vartype: '_Vartype',
+                            parameters: typing.MutableMapping[str, typing.Any],
+                            properties: typing.Mapping[str, typing.Any],
+                            inplace: bool = False,
+                            ) -> typing.MutableMapping[str, typing.Any]:
+        """Reformat some solver parameters for SAPI.
 
-                initial_state_list = [3]*self.properties['num_qubits']
+        Currently the only reformatted parameter is ``initial_state``. This
+        method allows ``initial_state`` to be submitted as a dictionary
+        mapping the qubits to their initial value.
 
-                low = -1 if type_ == 'ising' else 0
+        Args:
+            vartype: One of ``'ising'`` or ``'qubo'``. If :mod:`dimod` is
+                installed, this can also be any
+                :class:`~dimod.typing.VartypeLike`.
+            parameters: The parameters to submit to ths solver.
+            properties: The solver's properties. Note that this will
+                work with either :attr:`StructuredSolver.properties`
+                or :attr:`dwave.systems.DWaveSampler.properties`.
+
+        Returns:
+            The reformatted solver parameters.
+            If ``inplace`` this will be the ``parameters``, modified.
+            If ``not inplace`` then this will be a deep copy of ``parameters``,
+            with the relevant fields updated.
```

Comment:
```suggestion
            If ``inplace`` is True, this is the modified ``parameters``.
            If ``inplace`` is False, this is a deep copy of ``parameters``
            with the relevant fields updated.
```

Reply:
In code, yes, but here's `inplace` is a Boolean argument of the function, so we always use the "arg is True" & "arg is False" format (and never `not arg`).

Reply:
Ok, but why?

Reply:
So users can parse English and not Python: "if <varname> is <value>" is a grammatical clause, while "if <varname>" is a syntactically correct Python structure

Reply:
Yup, but here you're saying `inplace` (variable name) and not "in place" (preposition)

Reply:
Ok, I concede. And I see that they use this convention in the Python docs as well.

Reply:
Updated.